demos/CMakeLists.txt

```diff
@@ -12,7 +12,7 @@ string(TOUPPER ${BOARD} BOARD_U)
 if(NOT (TARGET SAMPLE::AZUREIOT))
     add_library(SAMPLE::AZUREIOT INTERFACE IMPORTED)
     target_sources(SAMPLE::AZUREIOT INTERFACE 
-        ${CMAKE_CURRENT_SOURCE_DIR}/sample_azure_iot/sample_azure_iot.c)
+        ${CMAKE_CURRENT_SOURCE_DIR}/sample_azure_iot_pnp/sample_azure_iot_pnp.c)
```

Comment:
how about we create one more target of PNP sample and let project decide, which one to pick based on some CMAKE variable pass?

Reply:
I was thinking this:

we create target 
SAMPLE::AZUREIOT
SAMPLE::AZUREIOT::PNP

and in project linux/st/nxp

we do selection of target linkage using this flag. 

nit: how about naming the variable as SAMPLE_DEMO?

Reply:
Ultimately I would like to just create executables for both. I don't think there's anything preventing us from creating both

Reply:
That is also fine... having target for both and letting executable to add dependency.

Reply:
done